doc/guides/device_mgmt/smp_protocol.rst

```diff
@@ -0,0 +1,289 @@
+    |                   |                                                   |
+    +-------------------+---------------------------------------------------+
+    | ``Data``          | This is data payload of the ``Data Length``       |
+    |                   | size. It is optional as ``Data Length`` may be    |
+    |                   | set to zero, which means that no data follows     |
+    |                   | the header.                                       |
+    |                   |                                                   |
+    +-------------------+---------------------------------------------------+
+
+.. note::
+    Although it may look strange that a frame begins with unused field (``Res``),
+    when the frame gets passed through UART/serial, by default, each byte
+    is sent in LSb to MSb order, which means that ``OP`` bits are sent first.
+
+
+    Contents of a ``Data`` depends on a value of an ``OP``, a ``Group ID``,
+    and a ``Command ID``.
+
+.. _mcumgr_smp_protocol_op_code:
+
+Operation code
+==============
+
+The operation code determines whether an information is written to a device or
+requested from it and whether a packet contains request to a SMP server or
+response from it.
+
+Following operation codes are defined
+
+.. table::
+    :align: center
+
+    +---------------+-----------------------------------------------+
+    | Decimal ID    | Operation                                     |
+    +===============+===============================================+
+    | ``0``         | write request                                 |
+    +---------------+-----------------------------------------------+
+    | ``1``         | write response                                |
+    +---------------+-----------------------------------------------+
+    | ``2``         | read request                                  |
+    +---------------+-----------------------------------------------+
+    | ``3``         | read response                                 |
+    +---------------+-----------------------------------------------+
+
+
+.. _mcumgr_smp_protocol_group_ids:
+
+Management ``Group ID``'s
+=========================
+
+The SMP protocol supports predefined common groups and allows user defined
+groups. Below table presents list of common groups:
+
+
+.. table::
+    :align: center
+
+    +---------------+-----------------------------------------------+
+    | Decimal ID    | Group description                             |
+    +===============+===============================================+
+    | ``0``         | :ref:`mcumgr_smp_group_0`                     |
+    +---------------+-----------------------------------------------+
+    | ``1``         | :ref:`mcumgr_smp_group_1`                     |
+    +---------------+-----------------------------------------------+
+    | ``2``         | :ref:`mcumgr_smp_group_2`                     |
+    +---------------+-----------------------------------------------+
```

Comment:
Does the name appear in these when built? If not, would add the name to these e.g. "OS management" etc.

Reply:
The name will be generated from chapters.